kubernetes/nodecollection_test.go

```diff
@@ -252,45 +290,65 @@ func TestDownloadNodeData(t *testing.T) {
 			t.Error("unexpected error")
 		}
 	})
+
+	t.Run("Ensure error is returned when GetReadyNodes returns error", func(t *testing.T) {
+		c := http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					// nolint gosec
+					InsecureSkipVerify: true,
+				},
+			}}
+		rc := raw.NewClient(
+			c,
+			true,
+			"",
+			uint(1),
+		)
+		ka := kubernetes.KubeAgentConfig{
+			Clientset:       cs,
+			HTTPClient:      c,
+			InClusterClient: rc,
+			ClusterHostURL:  "https://" + ts.Listener.Addr().String(),
+		}
+		wd, _ := os.Getwd()
+		ed, _ := os.Open(fmt.Sprintf("%s/testdata", wd))
+
+		ns := testNodeSource{}
+
+		_, err := kubernetes.DownloadNodeData(
+			"baseline",
+			ka,
+			ed,
+			ns,
+		)
+
+		if err == nil {
+			t.Error("Expected no nodes found error")
+		}
+
+		if err.Error() != "cloudability metric agent is unable to get a list of nodes: 0 nodes were ready" {
```

Comment:
If we ever change this error message, this test will fail. But at that point I guess we can just make it a constant.